File: src/test/java/com/tms/csp/csp/IsSatTest.java
package com.tms.csp.csp;

import com.tms.csp.ast.Csp;
import com.tms.csp.ast.Exp;
import com.tms.csp.data.CspSample;
import com.tms.csp.util.CspBaseTest2;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertTrue;

public class IsSatTest extends CspBaseTest2 {

    @Test
    public void testTiny() throws Exception {
        Csp csp = loadCsp(CspSample.Tiny);
        assertTrue(csp.isSat());
    }

    @Test
    public void testTrim() throws Exception {
        Csp csp = loadCsp(CspSample.Trim);
        assertTrue(csp.isSat());
    }

    @Test
    public void testTrimColor() throws Exception {
        Csp csp = loadCsp(CspSample.TrimColor);
        assertTrue(csp.isSat());
    }

    @Test
    public void testTrimColorOptions() throws Exception {
        Csp csp = loadCsp(CspSample.TrimColorOptions, true);
        List<Exp> xors = csp.getXorConstraints();
        for (Exp xor : xors) {
            System.err.println(xor);
        }
        assertTrue(csp.isSat());
    }

    @Test
    public void testCamry2011() throws Exception {
        Csp csp = loadCsp(CspSample.Camry2011, true);
        assertTrue(csp.isSat());
    }

    //591 millis
    @Test
    public void testEfc1() throws Exception {
        Csp csp = loadCsp(efcOriginal);
        long t1 = System.currentTimeMillis();
        assertTrue(csp.isSat());
        long t2 = System.currentTimeMillis();
        System.err.println("efc1.isSat Delta: " + (t2 - t1));
    }

    //859 millis
    @Test
    public void testEfc2() throws Exception {
        Csp csp = CspSample.EfcProdFactoryRules.csp();
        long t1 = System.currentTimeMillis();
        assertTrue(csp.isSat());
        long t2 = System.currentTimeMillis();
        System.err.println("efc2.isSat Delta: " + (t2 - t1));
    }

}

File: src/test/java/com/tms/csp/csp/IsSatTest.kt
package com.tms.csp.csp

import com.tms.csp.ast.Csp
import com.tms.csp.ast.HasIsSat
import com.tms.csp.data.CspSample
import org.junit.Assert.assertTrue
import org.junit.Test

class IsSatTest {

    fun parse(s: CspSample): HasIsSat {
        return Csp.parse(s)
    }

    @Test
    fun testTiny() {
        val csp = parse(CspSample.TinyDc)
        assertTrue(csp.isSat())
    }

    @Test
    fun testTrim() {
        val csp = parse(CspSample.TrimNoDc)
        assertTrue(csp.isSat())
    }

    @Test
    fun testTrimColor() {
        val csp = parse(CspSample.TrimColorNoDc)
        assertTrue(csp.isSat())
    }

    @Test
    fun testTrimColorOptions() {
        val csp = parse(CspSample.TrimColorOptionsDc)
        assertTrue(csp.isSat())
    }

    @Test
    fun testCamry2011() {
        val csp = parse(CspSample.Camry2011Dc)
        assertTrue(csp.isSat())
    }

    //591 millis
    @Test
    fun testEfc1() {
        val csp = parse(CspSample.EfcOriginal)
        val t1 = System.currentTimeMillis()
        assertTrue(csp.isSat())
        val t2 = System.currentTimeMillis()
        System.err.println("efc1.isSat Delta: " + (t2 - t1))
    }

    //859 millis
    @Test
    fun testEfc2() {
        val csp = parse(CspSample.EfcProdFactoryRules)
        val t1 = System.currentTimeMillis()
        assertTrue(csp.isSat())
        val t2 = System.currentTimeMillis()
        System.err.println("efc2.isSat Delta: " + (t2 - t1))
    }

}